shell/platform/windows/uwp_flutter_window.cc

```diff
@@ -0,0 +1,486 @@
+          cursor_visual_.Offset().y * inverseDpiScale,
+          FlutterPointerMouseButtons::kFlutterPointerButtonMousePrimary);
+    } else {
+      binding_handler_delegate_->OnPointerDown(
+          cursor_visual_.Offset().x, cursor_visual_.Offset().y,
+          FlutterPointerMouseButtons::kFlutterPointerButtonMousePrimary);
+    }
+  }
+}
+
+void WinRTFlutterWindow::OnGamePadButtonReleased(
+    winrt::Windows::Gaming::Input::GamepadButtons buttons) {
+  if ((buttons & winrt::Windows::Gaming::Input::GamepadButtons::A) ==
+      winrt::Windows::Gaming::Input::GamepadButtons::A) {
+    if (!running_on_xbox_) {
+      const double inverseDpiScale = GetDpiScale();
+
+      binding_handler_delegate_->OnPointerUp(
+          cursor_visual_.Offset().x * inverseDpiScale,
+          cursor_visual_.Offset().y * inverseDpiScale,
+          FlutterPointerMouseButtons::kFlutterPointerButtonMousePrimary);
+    } else {
+      binding_handler_delegate_->OnPointerUp(
+          cursor_visual_.Offset().x, cursor_visual_.Offset().y,
+          FlutterPointerMouseButtons::kFlutterPointerButtonMousePrimary);
+    }
+  }
+}
+
+void WinRTFlutterWindow::OnGamePadControllersChanged() {
+  // TODO lock here
+  if (game_pad_->HasController()) {
+    if (!game_controller_thread_running_) {
+      if (cursor_visual_ != nullptr) {
+        visual_tree_root_.Children().InsertAtTop(cursor_visual_);
+      }
+      StartGamepadCursorThread();
+      game_controller_thread_running_ = true;
+    } else {
+      if (cursor_visual_ != nullptr) {
+        visual_tree_root_.Children().Remove(cursor_visual_);
+      }
+      game_controller_thread_running_ = false;
+      // TODO stop game thread
+    }
+  }
+}
+
+winrt::Windows::UI::Composition::Visual
+WinRTFlutterWindow::CreateCursorVisual() {
+  auto container = compositor_.CreateContainerVisual();
+  container.Offset(
+      {window_.Bounds().Width / 2, window_.Bounds().Height / 2, 1.0});
+
+  const float SIZE = 30;
+  auto cursor_visual = compositor_.CreateShapeVisual();
+  cursor_visual.Size({SIZE, SIZE});
+
+  // compensate for overscan in cursor visual
+  cursor_visual.Offset({xbox_overscan_x_offset_, xbox_overscan_y_offset_, 1.0});
+
+  winrt::Windows::UI::Composition::CompositionEllipseGeometry circle =
+      compositor_.CreateEllipseGeometry();
+  circle.Radius({SIZE / 2, SIZE / 2});
+
+  auto circleshape = compositor_.CreateSpriteShape(circle);
+  circleshape.FillBrush(
+      compositor_.CreateColorBrush(winrt::Windows::UI::Colors::Black()));
+  circleshape.Offset({SIZE / 2, SIZE / 2});
+
+  cursor_visual.Shapes().Append(circleshape);
+
+  winrt::Windows::UI::Composition::Visual visual =
+      cursor_visual.as<winrt::Windows::UI::Composition::Visual>();
+
+  visual.CompositeMode(winrt::Windows::UI::Composition::
+                           CompositionCompositeMode::DestinationInvert);
+
+  visual.AnchorPoint({0.5, 0.5});
+  container.Children().InsertAtTop(visual);
+
+  return container;
+}
+
```

Comment:
Could you do a pass over this file's TODOs? It's not clear which ones are intended to actually be future work, and which were notes to yourself during prototyping. For the former, they could use more details so that it's clear what the remaining work is.

Reply:
Done: If removed bogus ones and hooked up the legit ones to the relevant issues, creating new issues if necessary.